ping/ping.c

```diff
@@ -362,6 +362,17 @@ main(int argc, char **argv)
 	else if (argv[0][strlen(argv[0]) - 1] == '6')
 		hints.ai_family = AF_INET6;
 
+	/*
+	 * Optionally disable reverse DNS resolution (PTR lookup) by default.
+	 * -n/-H override the variable.
+	 */
+	char *env = getenv("IPUTILS_PING_PTR_LOOKUP");
+	if (env && !strcmp(env, "0")) {
+		rts.opt_numeric = 1;
+		if (rts.opt_verbose)
+			error(0, 0, _("WARNING: reverse DNS resolution (PTR lookup) disabled, enforce with -H"));
```

Comment:
So we are checking the `opt_verbose` before the getopt() loop that sets it?

Reply:
Thank you, I'll move it below.

Reply:
Warning moved below, don't print on `-q` (enabled by default, no need to put `-v`).